### PR TITLE
Adding eventDate to event model

### LINKED ISSUE #121: Implementing update function of events for admin
### Description
Complete CRUD for events by adding updating functionality.
Updating functionality will require maintaining a history of updates!

### Screenshots/testimonials
<place your screenshots or testimonials here if applicable>
N/A

### LINKED ISSUE #167: Add eventDate in the event model
### Description

With reference to #116, backend should also be serving the event dates for it to be displayed on the event component. This should be defaulted to `Date.now()` and must be required for every event. We may also have the functionality of storing `endDate` to be more precise about event tenures etc. This is to be discussed further.

### Screenshots/testimonials
NA

### PR DESCRIPTION
**Pull request by EmperorYP7**
fixes #167 
fixes #121 
<Do not delete this template>
Description: Added the eventDate field to event model. General clean up and added auth protected route for updating event details.

#### Checklist
- [x] I've followed the Contributing guidelines provided in the repository.
- [x] I've made the changes which were demanded in the linked issue.
- [ ] I've tested my code on a Chromium based browser.
- [ ] I've tested my code on Mozilla Firefox.
- [ ] My code gave a clean console on debugging. (no warnings/errors)

#### Testimonial
NA

### CODE REVIEW

File: backend/routes/event.js [hunk 1]
```diff
@@ -85,9 +86,29 @@ router.delete("/:id", auth, (req, res, next) => {
...
-  res.status(500).json('bad request or 404 not found');
-});
+// router.get('*', function(req, res){
+//   res.status(500).json('bad request or 404 not found');
+// });
+
 // Should we allow updation to events, requires disussion
 
+router.put("/:id", auth, (req, res) => { 
+    const event = Event.findOne({ _id: req.params.id })
```
Comment (anktrj) [inline backend/routes/event.js:92]: use findOneAndUpdate instead
also, add a boolean to the schema named isUpdated and set it to true here (default false)

File: backend/models/event.model.js [hunk 2]
```diff
@@ -16,6 +16,11 @@ const eventSchema = new mongoose.Schema({
         type: Date,
         default: Date.now
     },
+    eventDate: {
```
Comment (anktrj) [inline backend/models/event.model.js:19]: Add a boolean named isUpdated with a default value false

**Review by ankiiitraj — CHANGES_REQUESTED** [review]

**Comment by ankiiitraj** [comment]
Remember to update the PR title

File: backend/models/event.model.js (hunk 2 above)
Comment (EmperorYP7) [reply backend/models/event.model.js:19]: Done.

File: backend/routes/event.js (hunk 1 above)
Comment (EmperorYP7) [reply backend/routes/event.js:92]: Tested it with `findOneAndUpdate`, there was some retundency so, settled for this.

File: backend/routes/event.js [hunk 3]
```diff
@@ -87,4 +88,24 @@ router.delete("/:id", auth, (req, res, next) => {
...
+                        .then(() => {
+                            event.isUpdated = true;
+                            event.title = req.body.title;
+                            event.community = req.body.community;
+                            event.description = req.body.description;
+                            event.pdfPath = req.body.pdfPath;
+                            event.eventLink = req.body.eventLink;
+                            event.eventDate = req.body.eventDate;
+                            event.banner = req.body.banner;
```
Comment (anktrj) [inline backend/routes/event.js:100]: You forgot to return the response!
return this event with a message OK

**Review by ankiiitraj — CHANGES_REQUESTED** [review]

File: backend/routes/event.js (hunk 3 above)
Comment (EmperorYP7) [reply backend/routes/event.js:100]: Woops. Completely forgot this. Thanks!

**Review by ankiiitraj — APPROVED** [review]

**Pull request merged**